### ISSUE TITLE
Sink Page: Modal suppress Enter issue

### ISSUE THREAD

**Issue by jbrcna** [infor-design/enterprise#5990]
**Describe the bug**
Enter key is not suppressed when you open and close (hit enter) both of the modal, one after the other
Open/Close the 2nd modal using Enter
Open/Close the 1st modal using Enter
Open/Close the 2nd modal using Enter. key not suppressed

**To Reproduce**

1. Go to https://4590-beta0-enterprise.demo.design.infor.com/components/modal/example-index.html
2. Move to modal element 
3. Click `Show Modal (Suppressed Enter)` to open the modal
4. Hit` Enter `key
5. Click `Show Modal ` to open the modal
6. Hit `Enter` key
7. Click `Show Modal (Suppressed Enter)` to open the modal
8. Hit `Enter` key
9. See error
10. Notice that enter key is not suppressed anymore.

**Expected behavior**
Enter key should be suppressed.

**Version**

- ids-enterprise v4.59.0-beta0

**Screenshots**
![Large GIF (854x540)](https://user-images.githubusercontent.com/56722184/148179937-e3d6b343-7be0-4d8f-983b-ec9d1ac71a84.gif)

**Platform**
all browsers

**Additional context**

**Comment by tmcconechy** [comment]
This one is a bit wierd? I only saw it once. Might be due to hitting enter too fast? Any other tips to reproduced this @jbrcna ?

**Comment by jbrcna** [comment]
I did both (fast and normal) press and the issue persists. below are the steps 

1. Go to https://4590-beta0-enterprise.demo.design.infor.com/components/modal/example-index.html
2. Click `Show Modal (Suppressed Enter)` to open the modal
3. Press `Enter` key. working as expected
4. Close the modal either with `Escape` key or `Cancel/Save` button
5. Click `Show Modal ` (1st modal) to open the modal
6. Press `Enter` key or` Cancel/Save` button
7. Click `Show Modal (Suppressed Enter)` again
8. Press `Enter` key. the enter key is not suppressed.

I noticed that on `step 6` if you press `Escape` key instead of `Enter` key or` Cancel/Save` button
then you open the suppressed modal again . it is working as expected, the enter key is suppressed.
to replicate the issue, try pressing `Enter` key or` Cancel/Save` button when closing the `Show Modal ` (1st modal)
